src/python/WMCore/WMSpec/StdSpecs/StdBase.py

```diff
@@ -391,12 +391,14 @@ def setupProcessingTask(self, procTask, taskType, inputDataset=None, inputStep=N
         procTaskCmsswHelper.setOverrideCatalog(self.overrideCatalog)
         procTaskCmsswHelper.setErrorDestinationStep(stepName=procTaskLogArch.name())
 
-        if forceMerged:
+        if forceMerged and isinstance(forceMerged, bool):
             procTaskStageHelper.setMinMergeSize(0, 0)
-        elif forceUnmerged:
+        elif forceUnmerged and isinstance(forceUnmerged, bool):
             procTaskStageHelper.disableStraightToMerge()
         else:
             procTaskStageHelper.setMinMergeSize(self.minMergeSize, self.maxMergeEvents)
+            if forceUnmerged and isinstance(forceUnmerged, list):
```

Comment:
1 is discussed above. 2 is the correct behavior. If we go into the elif at 396, it's because forceUnmerged is a boolean and is True. There is no need to go into the else, we just want to disable all straight to merge during stageout.

The else is only triggered if neither forceMerged and forceUnmerged is True, which triggers the normal case of straight to merge enabled with the usual thresholds. Only in that case do I want to allow an override for specific output modules that should always write unmerged.